satpy/resample.py

```diff
@@ -509,27 +472,117 @@ def compute(self, data, fill_value=None, **kwargs):
+                           for a in range(d_arr.ndim))
+            return d_arr[slices]
+
+    def compute(self, data, expand=True, **kwargs):
+        if isinstance(self.target_geo_def, (list, tuple)):
+            # find the highest/lowest area among the provided
+            test_func = max if expand else min
+            target_geo_def = test_func(self.target_geo_def,
+                                       key=lambda x: x.shape)
+        else:
+            target_geo_def = self.target_geo_def
+
+        # convert xarray backed with numpy array to dask array
+        if 'x' not in data.dims or 'y' not in data.dims:
+            if data.ndim not in [2, 3]:
+                raise ValueError("Can only handle 2D or 3D arrays without "
+                                 "dimensions.")
+            # assume rows is the second to last axis
+            y_axis = data.ndim - 2
+            x_axis = data.ndim - 1
+        else:
+            y_axis = data.dims.index('y')
+            x_axis = data.dims.index('x')
+
+        out_shape = target_geo_def.shape
+        in_shape = data.shape
+        y_repeats = out_shape[y_axis] / float(in_shape[y_axis])
+        x_repeats = out_shape[x_axis] / float(in_shape[x_axis])
+
+        d_arr = self.expand_reduce(data.data, y_repeats, y_axis)
+        d_arr = self.expand_reduce(d_arr, x_repeats, x_axis)
+
+        coords = {}
+        # Update coords if we can
+        if 'y' in data.coords or 'x' in data.coords and \
+                isinstance(target_geo_def, AreaDefinition):
+            coord_chunks = (d_arr.chunks[y_axis], d_arr.chunks[x_axis])
+            x_coord, y_coord = target_geo_def.get_proj_vectors_dask(
+                chunks=coord_chunks)
+            if 'y' in data.coords:
+                coords['y'] = y_coord
+            if 'x' in data.coords:
+                coords['x'] = x_coord
+
+        return xr.DataArray(d_arr,
+                            dims=data.dims,
+                            coords=coords or None)
+
+
 RESAMPLERS = {"kd_tree": KDTreeResampler,
               "nearest": KDTreeResampler,
               "ewa": EWAResampler,
               "bilinear": BilinearResampler,
+              "native": NativeResampler,
               }
 
 RESAMPLER_CACHE = WeakValueDictionary()
 
 
 def resample(source_area, data, destination_area,
-             resampler_class=KDTreeResampler, **kwargs):
+             resampler=None, **kwargs):
     """Do the resampling."""
-    key = (source_area, destination_area)
+    if 'resampler_class' in kwargs:
+        import warnings
+        warnings.warn("'resampler_class' is deprecated, use 'resampler'",
+                      DeprecationWarning)
+        resampler = kwargs.pop('resampler_class')
 
-    resampler = RESAMPLER_CACHE.get(key)
+    key = (source_area, destination_area)
+    if key in RESAMPLER_CACHE:
+        resampler = RESAMPLER_CACHE[key]
+    else:
+        if resampler is None:
+            resampler = 'kd_tree'
```

Comment:
we should add at least a log info that kd_tree is chosen since nothing else was provided